include/Executor.hpp

```diff
@@ -0,0 +1,17 @@
+//
+// Created by Tianyi Wang on 5/14/23.
+//
+
+#ifndef DD_EVAL_EXECUTOR_HPP
+#define DD_EVAL_EXECUTOR_HPP
+
+#endif // DD_EVAL_EXECUTOR_HPP
```

Comment:
These are called header guards. Code should go in between the `#define ...` and the `#endif`. Otherwise they don't serve a purpose at all.
However, it's actually the more modern practice to use `#pragma once` instead.
As for the file here, that would be:
```suggestion
#pragma once
```
Please also adapt the other files accordingly.